farm/data_handler/processor.py

```diff
@@ -1671,6 +1672,168 @@ def _sample_to_features(self, sample) -> dict:
+        :param metric: name of metric that shall be used for evaluation, e.g. "acc" or "f1_macro".
+                 Alternatively you can also supply a custom function, that takes preds and labels as args and returns a numerical value.
+                 For using multiple metrics supply them as a list, e.g ["acc", my_custom_metric_fn].
+        :type metric: str, function, or list
+        :param train_filename: The name of the file containing training data.
+        :type train_filename: str
+        :param dev_filename: The name of the file containing the dev data. If None and 0.0 < dev_split < 1.0 the dev set
+                             will be a slice of the train set.
+        :type dev_filename: str or None
+        :param test_filename: None
+        :type test_filename: str
+        :param dev_split: The proportion of the train set that will sliced. Only works if dev_filename is set to None
+        :type dev_split: float
+        :param proxies: proxy configuration to allow downloads of remote datasets.
+                        Format as in  "requests" library: https://2.python-requests.org//en/latest/user/advanced/#proxies
+        :type proxies: dict
+        :param embed_title: Whether to embed title in passages during tensorization (bool),
+        :param num_hard_negatives: maximum number to hard negative context passages in a sample
+        :param num_positives: maximum number to positive context passages in a sample
+        :param kwargs: placeholder for passing generic parameters
+        :type kwargs: object
+        """
+        #TODO If an arg is misspelt, e.g. metrics, it will be swallowed silently by kwargs
+
+        # Custom processor attributes
+        self.max_samples = max_samples
+        self.query_tokenizer = query_tokenizer
+        self.passage_tokenizer = passage_tokenizer
+        self.embed_title = embed_title
+        self.num_hard_negatives = num_hard_negatives
+        self.num_positives = num_positives
+
+        super(DPRProcessor, self).__init__(
+            tokenizer= query_tokenizer,
+            max_seq_len=max_seq_len,
+            train_filename=train_filename,
+            dev_filename=dev_filename,
+            test_filename=test_filename,
+            dev_split=dev_split,
+            data_dir=data_dir,
+            tasks={},
+            proxies=proxies,
+        )
+        if metric:
+            task_type = "dpr"
+            self.add_task(name="dpr",
+                          metric=metric,
+                          label_list=None,
+                          label_name=None,
+                          task_type=task_type)
+        else:
+            logger.info("Initialized processor without tasks. Supply `metric` and `label_list` to the constructor for "
+                        "using the default task or add a custom task later via processor.add_task()")
+
+    def file_to_dicts(self, file: str) -> [dict]:
+        # TO-DO: json should be in SQUAD format
```

Comment:
I think original DPR format as input is okay for now as negative passages would be part to represent in SQuAD format. 
However, we should convert the dicts that we read in here to the format that we expect in dict_to_samples

Reply:
changed it to similar format